esmvaltool/preprocessor/_multimodel.py

```diff
@@ -164,133 +177,73 @@ def _get_overlap(cubes):
     """
     all_times = []
     for cube in cubes:
-        # monthly data ONLY
-        # converts all to 365-days calendars
-        time_units = cube.coord('time').units
-        bnd1 = float(cube.coord('time').points[0])
-        bnd2 = float(cube.coord('time').points[-1])
-        if time_units.calendar is not None:
-            if time_units.calendar == '360_day':
-                bnd1 = int(bnd1 / 360.) * 365.
-                bnd2 = (int(bnd2 / 360.) + 1) * 365.
-            else:
-                bnd1 = int(bnd1 / 365.) * 365.
-                bnd2 = (int(bnd2 / 365.) + 1) * 365.
-        else:
-            logger.debug("No calendar type: assuming 365-day")
-            bnd1 = int(bnd1 / 365.) * 365.
-            bnd2 = (int(bnd2 / 365.) + 1) * 365.
+        bnd1 = _datetime_to_int_days(cube)[0]
+        bnd2 = _datetime_to_int_days(cube)[-1]
         all_times.append([bnd1, bnd2])
-    bounds = [range(int(b[0]), int(b[-1]) + 1) for b in all_times]
+    bounds = [range(b[0], b[-1] + 1) for b in all_times]
     time_pts = reduce(np.intersect1d, bounds)
     if len(time_pts) > 1:
         time_bounds_list = [time_pts[0], time_pts[-1]]
         return time_bounds_list
 
 
-def _slice_cube(cube, min_t, max_t):
-    """slice cube on time"""
-    fmt = '%Y-%m-%d-%H'
-    ctr = iris.Constraint(time=lambda x:
-                          min_t <=
-                          datetime.strptime(x.point.strftime(fmt),
-                                            fmt) <= max_t)
-    cube_slice = cube.extract(ctr)
-    return cube_slice
-
-
-def _slice_cube2(cube, t_1, t_2):
+def _slice_cube(cube, t_1, t_2):
     """
     Efficient slicer
 
-    Slice cube on time more memory-efficiently than
-    _slice_cube(); using this function adds virtually
+    Cube indexing has a large memory leak!
```

Comment:
That seems unlikely, you probably mean something else.

Reply:
yeah, my docstring was vague as hell; I meant indexing cubes and keeping the indexed cubes in a multimodel list of cubes is taking a fair bit of memory